portal/auth.py

```diff
@@ -0,0 +1,94 @@
+from flask import (
+    Blueprint, flash, g, redirect, render_template, request, session, url_for
+)
+import functools
+from . import db
+
+bp = Blueprint("auth", __name__)
+
+@bp.route('/login', methods=('GET', 'POST'))
+def login():
+    """Renders the login page.
+    Starts a new session if a login attempt is successful."""
+
+    if request.method == 'POST':
+
+        con = db.get_db()
+        cur = con.cursor()
+
+        email = request.form['email']
+        password = request.form['password']
+
+        cur.execute('SELECT * FROM users WHERE email = %s', (email,))
+        user = cur.fetchone()
```

Comment:
Swap order of these steps:
1. get form data
2. open connection and cursor
3. run queries and fetch data
4. close connection and cursor just in case of errors
5. process form...